### ISSUE TITLE
Handling DI annotations in `resolve`

### ISSUE THREAD

**Issue by Alexander3** [angular-ui/ui-router#3387]
This is a:

- [x] Bug Report
- [ ] Feature Request
- [ ] General Query 

My version of UI-Router is: 0.4.2 
and I'm using: "angular": "~1.5"

# Bug Report

#### Current Behavior:

**resolve** state parameter can't handle dependency injection properly, it fails when code is minified.
This was extremely hard to debug. Generally I was trying to check if state params are present, like someone suggested  [here](http://stackoverflow.com/questions/37758274/angularjs-ui-router-mandatory-parameters)

#### Expected Behavior:

DI annotations should be handled consistently with the rest of angular. I am aware that this is probably bad usage of **resolve** , but if it allows DI it should handle all cases.

#### Link to Plunker that reproduces the issue:

https://plnkr.co/edit/RnLcUc5h6zbPvGjkIo0P?p=preview

**Comment by christopherthielen** [comment]
No, you aren't using `resolve` correctly.  Resolve is an object with keys and values.

Your example should read something like:
```js
      resolve: { 
        somekey: ['$stateParams', '$location', function($stateParamsMinified, $location) {
         console.log("This won't work for minified arguments", $stateParamsMinified, $location.path());
        }] 
      }
```

**Comment by Alexander3** [comment]
@christopherthielen Ok, but why this is working ?? It should give me warning or sth.
```
resolve: function($stateParams, $location) {
         console.log($stateParams, $location.path());
      }
```

**Comment by christopherthielen** [comment]
it probably works because of a bug or something.  it's been fixed in 1.0 branch.  Yeah a warning might be nice for obviously invalid resolve: block.  It's not high on my list, so PR welcomed

**Comment by Alexander3** [comment]
@christopherthielen https://github.com/angular-ui/ui-router/pull/3388
